examples/text-classification/run_glue.py

```diff
@@ -571,162 +587,169 @@ def take_train_steps(model, trainer, resume_from_checkpoint, last_checkpoint):
 
-        if teacher_tokenizer.vocab != tokenizer.vocab:
-            raise ValueError("Teacher model and student model should have same tokenizer.")
+        # Save the resulting model and its corresponding configuration in the given directory
+        optimizer.save_pretrained(training_args.output_dir)
+        # Save the tokenizer for reloading
+        tokenizer.save_pretrained(training_args.output_dir)
+        # Compute the model's sparsity
+        sparsity = optimizer.get_sparsity()
 
-        distillation_config = IncDistillationConfig.from_pretrained(
-            optim_args.distillation_config if optim_args.distillation_config is not None else default_config,
-            config_file_name="distillation.yml",
-            cache_dir=model_args.cache_dir,
+        logger.info(
+            f"Optimized model with {metric_name} of {result_optimized_model} and sparsity of {round(sparsity, 2)}% "
+            f"saved to: {training_args.output_dir}. Original model had an {metric_name} of {result_baseline_model}."
         )
 
-        # Creation Distillation object used for IncTrainer training loop
-        distiller = IncDistiller(
-            teacher_model=teacher_model, config=distillation_config, eval_func=eval_func, train_func=train_func
-        )
-
-    optimizer = IncOptimizer(
-        model,
-        quantizer=quantizer,
-        pruner=pruner,
-        distiller=distiller,
-        one_shot_optimization=True,
-        eval_func=eval_func,
-        train_func=train_func,
-    )
-
-    agent = optimizer.get_agent()
-    optimized_model = optimizer.fit()
-    result_optimized_model = take_eval_steps(optimized_model, trainer, metric_name, save_metrics=True)
-
-    # Save the resulting model and its corresponding configuration in the given directory
-    optimizer.save_pretrained(training_args.output_dir)
-    # Compute the model's sparsity
-    sparsity = optimizer.get_sparsity()
-
-    logger.info(
-        f"Optimized model with {metric_name} of {result_optimized_model} and sparsity of {round(sparsity, 2)}% "
-        f"saved to: {training_args.output_dir}. Original model had an {metric_name} of {result_baseline_model}."
-    )
-
-    if optim_args.apply_quantization and optim_args.verify_loading:
+    if optim_args.verify_loading:
 
         # Load the model obtained after Intel Neural Compressor quantization
-        loaded_model = IncQuantizedModelForSequenceClassification.from_pretrained(training_args.output_dir)
+        if optim_args.load_int8:
+            loaded_model = model
+        else:
+            loaded_model = IncQuantizedModelForSequenceClassification.from_pretrained(training_args.output_dir)
         loaded_model.eval()
+
         result_loaded_model = take_eval_steps(loaded_model, trainer, metric_name)
 
-        if result_loaded_model != result_optimized_model:
+        if not optim_args.load_int8 and result_loaded_model != result_optimized_model:
```

Comment:
Not sure about the case where `optim_args.load_int8` is set to `True` as we don't have the original model's accuracy to compare it with

Reply:
Yes, I also have this concern. One possible method is checking the module class whether there is 'quantize' string in its name.

Reply:
We could create a separate function verifying the model's quantization by comparing what layer is being quantized with the configuration describing the quantization process, but it might be a little bit heavy